src/libstore/globals.hh

```diff
@@ -14,6 +14,12 @@ namespace nix {
 
 typedef enum { smEnabled, smRelaxed, smDisabled } SandboxMode;
 
+NLOHMANN_JSON_SERIALIZE_ENUM(SandboxMode, {
```

Comment:
Doesn't this require globals.hh to include nlohmann/json.hpp? So it wouldn't work with `PRECOMPILE_HEADERS=0`.

Reply:
I think it does because it passed tests? That header is included via `abstractsettingtojson.hh`

Reply:
Ouch, we need to revert that. It causes most compilation units to pull in nlohmann/json.hpp.

Reply:
Background: including that header adds about half a second per compilation unit.

Reply:
I've rebased this on master to incorporate your removal of `abstract-setting-to-json.hh` from `globals.hh` and moved the macro call to `globals.cc`, which I hope is satisfactory.

I tested locally that this still works for manpage-generation purposes.